server/gitrest/packages/gitrest-base/src/utils/redisFs/redisFsManager.ts

```diff
@@ -0,0 +1,457 @@
+	 * isomorphic-git never provides options, and always expects string[] results.
+	 */
+	public async readdir(
+		folderpath: PathLike,
+		options?:
+			| (ObjectEncodingOptions & { withFileTypes?: false | undefined })
+			| BufferEncoding
+			| null,
+	): Promise<string[]>;
+	public async readdir(
+		folderpath: PathLike,
+		options: { encoding: "buffer"; withFileTypes?: false | undefined } | "buffer",
+	): Promise<Buffer[]>;
+	public async readdir(
+		folderpath: PathLike,
+		options?:
+			| (ObjectEncodingOptions & { withFileTypes?: false | undefined })
+			| BufferEncoding
+			| null,
+	): Promise<string[] | Buffer[]>;
+	public async readdir(
+		folderpath: PathLike,
+		options: ObjectEncodingOptions & { withFileTypes: true },
+	): Promise<Dirent[]>;
+	public async readdir(
+		folderpath: PathLike,
+		options?: any,
+	): Promise<string[] | Buffer[] | Dirent[]> {
+		const folderpathString = folderpath.toString();
+
+		const result = await executeRedisFsApi(
+			async () => this.redisFs.keys_by_prefix(folderpath.toString()),
+			RedisFsApis.Readdir,
+			RedisFSConstants.RedisFsApi,
+			this.redisFsConfig.enableRedisFsMetrics,
+			this.redisFsConfig.redisApiMetricsSamplingPeriod,
+			{
+				folderpathString,
+			},
+		);
+
+		return result;
+	}
+
+	/**
+	 * Asynchronously creates a directory.
+	 * For the redis implementation, the subdiretories are not useful.
+	 * Therefore, we ignore the recursive flag.
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L115
+	 */
+	public async mkdir(
+		folderpath: PathLike,
+		options: MakeDirectoryOptions & { recursive: true },
+	): Promise<string | undefined>;
+	public async mkdir(
+		folderpath: PathLike,
+		options?: Mode | (MakeDirectoryOptions & { recursive?: false | undefined }) | null,
+	): Promise<void>;
+	public async mkdir(
+		path: PathLike,
+		options?: Mode | MakeDirectoryOptions | null,
+	): Promise<string | undefined>;
+	public async mkdir(
+		folderpath: PathLike,
+		options?: Mode | MakeDirectoryOptions | null,
+	): Promise<undefined | string | void> {
+		const folderpathString = folderpath.toString();
+
+		await executeRedisFsApi(
+			async () => this.redisFs.set(folderpath.toString(), ""),
+			RedisFsApis.Mkdir,
+			RedisFSConstants.RedisFsApi,
+			this.redisFsConfig.enableRedisFsMetrics,
+			this.redisFsConfig.redisApiMetricsSamplingPeriod,
+			{
+				folderpathString,
+			},
+		);
+	}
+
+	/**
+	 * Removes the directory identified by folderpath. For removing subdirectories and files,
+	 * `rm` should be used instead.
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L152
+	 */
+	public async rmdir(folderpath: PathLike, options?: RmDirOptions): Promise<void> {
+		const keys_to_remove = await this.redisFs.keys_by_prefix(folderpath.toString());
```

Comment:
Any reason for not wrapping up these async calls within a `await executeRedisFsApi(...)`?